governance/program/tests/process_remove_token_owner_record_lock.rs

```diff
@@ -0,0 +1,266 @@
+    // Arrange
+    let mut governance_test = GovernanceProgramTest::start_new().await;
+
+    let realm_cookie = governance_test.with_realm().await;
+
+    let token_owner_record_cookie = governance_test
+        .with_community_token_deposit(&realm_cookie)
+        .await
+        .unwrap();
+
+    let token_owner_record_lock_authority_cookie = governance_test
+        .with_community_token_owner_record_lock_authority(&realm_cookie)
+        .await
+        .unwrap();
+
+    let token_owner_record_lock_cookie = governance_test
+        .with_token_owner_record_lock(
+            &token_owner_record_cookie,
+            &token_owner_record_lock_authority_cookie,
+        )
+        .await
+        .unwrap();
+
+    // Act
+    let err = governance_test
+        .remove_token_owner_record_lock_using_ix(
+            &token_owner_record_cookie,
+            &token_owner_record_lock_authority_cookie.authority,
+            token_owner_record_lock_cookie.lock_type,
+            |i| i.accounts[1].is_signer = false,
+            Some(&[]),
+        )
+        .await
+        .err()
+        .unwrap();
+
+    // Assert
+    assert_eq!(
+        err,
+        GovernanceError::TokenOwnerRecordLockAuthorityMustSign.into()
+    );
+}
+
+#[tokio::test]
+async fn test_remove_token_owner_record_lock_after_authority_revoked() {
+    // Arrange
+    let mut governance_test = GovernanceProgramTest::start_new().await;
+
+    let realm_cookie = governance_test.with_realm().await;
+
+    let token_owner_record_cookie = governance_test
+        .with_community_token_deposit(&realm_cookie)
+        .await
+        .unwrap();
+
+    let token_owner_record_lock_authority_cookie = governance_test
+        .with_community_token_owner_record_lock_authority(&realm_cookie)
+        .await
+        .unwrap();
+
+    let token_owner_record_lock_cookie = governance_test
+        .with_token_owner_record_lock(
+            &token_owner_record_cookie,
+            &token_owner_record_lock_authority_cookie,
+        )
+        .await
+        .unwrap();
+
+    // Revoke authority
+    let args = SetRealmConfigItemArgs::TokenOwnerRecordLockAuthority {
+        action: SetConfigItemActionType::Remove,
+        governing_token_mint: realm_cookie.account.community_mint,
+        authority: token_owner_record_lock_authority_cookie.authority.pubkey(),
+    };
+
+    governance_test
+        .set_realm_config_item(&realm_cookie, args)
+        .await
+        .unwrap();
+
+    // Act
+    governance_test
+        .remove_token_owner_record_lock(
+            &token_owner_record_cookie,
+            &token_owner_record_lock_authority_cookie.authority,
+            token_owner_record_lock_cookie.lock_type,
+        )
+        .await
+        .unwrap();
+
+    // Assert
+    let token_owner_record_account = governance_test
+        .get_token_owner_record_account(&token_owner_record_cookie.address)
+        .await;
+
+    assert_eq!(0, token_owner_record_account.locks.len());
+}
```

Comment:
My original thinking was to support a scenario when a `Realm` is reconfigured with a new `plugin` or `authority` and it might  be required  to remove the locks issued by the previous `plugin` or `authority`.
I realised the solution was not good enough because it would still require the old `plugin` or `authority` signature which was not really guaranteed to work and this is why I modified `RelinquishTokenOwnerRecordLocks` to make it possible to remove the old locks without the revoked authority signature. 
This way it also covers the scenario with the rouge authority where it could be removed and its locks relinquished permissionless from `TokenOwnerRecords`